### DISCUSSION TITLE
How to obtain all the styles, tools, and components？

### DISCUSSION THREAD

**Discussion by hcweb** [tailwindlabs/tailwindcss#11539] [Help]
How to obtain all the styles, tools, and components, similar to bootstrap. After downloading, there will be a bootstrap. min.css, which is currently used https://cdn.tailwindcss.com But it keeps failing to load?

**Comment by wongjn** [comment]
This would not really be possible for v3 – there's an infinite amount of classes possible (due to the arbitrary features) and even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (70![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.

[^1]: 70 factorial, equal to 11978571669969891796072783721689098736458938142546425857555362864628009582789845319680000000000000000
